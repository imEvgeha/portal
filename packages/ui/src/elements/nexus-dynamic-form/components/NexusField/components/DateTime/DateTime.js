import React from 'react';
import PropTypes from 'prop-types';
import NexusDatePicker from '../../../../../nexus-date-and-time-elements/nexus-date-picker/NexusDatePicker';
import NexusDateTimePicker from '../../../../../nexus-date-and-time-elements/nexus-date-time-picker/NexusDateTimePicker';
import NexusDateTimeWindowPicker from '../../../../../nexus-date-and-time-elements/nexus-date-time-window-picker/NexusDateTimeWindowPicker';
import {DATETIME_FIELDS} from '../../../../constants';
import './DateTime.scss';

const {BUSINESS_DATETIME, TIMESTAMP} = DATETIME_FIELDS;

const DateTime = ({
    dateType,
    type,
    labels,
    isReadOnly,
    isDisabled,
    value,
    onChange,
    id,
    isReturningTime,
    isClearable,
}) => {
    const isUsingTime = [BUSINESS_DATETIME, TIMESTAMP].includes(dateType);
    const isTimestamp = dateType === TIMESTAMP;
    if (type === 'dateRange') {
        const dateProps = {
            isUsingTime,
            isTimestamp,
            startDateTimePickerProps: {
                id: `${id}-start`,
                defaultValue: value.startDate,
            },
            endDateTimePickerProps: {
                id: `${id}-end`,
                defaultValue: value.endDate,
            },
            labels,
            onChange,
            isReturningTime,
            isClearable,
        };
        return <NexusDateTimeWindowPicker isReadOnly={isReadOnly || isDisabled} {...dateProps} />;
    }
    const props = {
        id: `${id}-date`,
        isReadOnly,
        value,
        isTimestamp,
        onChange,
        isClearable,
    };
    return isUsingTime ? (
        <NexusDateTimePicker {...props} value={value.toString()} />
    ) : (
        <NexusDatePicker {...props} value={value.toString()} />
    );
};

DateTime.propTypes = {
    dateType: PropTypes.string,
    type: PropTypes.string,
    id: PropTypes.string,
    labels: PropTypes.array,
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
    isReadOnly: PropTypes.bool,
    isDisabled: PropTypes.bool,
    isClearable: PropTypes.bool,
    onChange: PropTypes.func,
    isReturningTime: PropTypes.bool,
};

DateTime.defaultProps = {
    dateType: '',
    type: '',
    id: '',
    labels: ['', ''],
    value: '',
    isReadOnly: false,
    isClearable: false,
    isDisabled: false,
    isReturningTime: true,
    onChange: () => null,
};

export default DateTime;
