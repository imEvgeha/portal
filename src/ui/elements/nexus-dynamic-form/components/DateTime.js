import React from 'react';
import PropTypes from 'prop-types';
import NexusDateTimeWindowPicker from '../../nexus-date-and-time-elements/nexus-date-time-window-picker/NexusDateTimeWindowPicker';
import {DATETIME_FIELDS} from '../constants';

const {BUSINESS_DATETIME, TIMESTAMP} = DATETIME_FIELDS;

const DateTime = ({dateType, labels, isReadOnly, value, onChange}) => {
    const dateProps = {
        isUsingTime: [BUSINESS_DATETIME, TIMESTAMP].includes(dateType),
        isTimestamp: dateType === TIMESTAMP,
        startDateTimePickerProps: {
            id: labels[0].replace(' ', ''),
            defaultValue: value.startDate,
        },
        endDateTimePickerProps: {
            id: labels[1].replace(' ', ''),
            defaultValue: value.endDate,
        },
        labels,
        onChange,
    };
    return <NexusDateTimeWindowPicker isReadOnly={isReadOnly} {...dateProps} />;
};

DateTime.propTypes = {
    dateType: PropTypes.string,
    labels: PropTypes.array,
    value: PropTypes.array,
    isReadOnly: PropTypes.bool,
    onChange: PropTypes.func.isRequired,
};

DateTime.defaultProps = {
    dateType: '',
    labels: ['', ''],
    value: ['', ''],
    isReadOnly: false,
};

export default DateTime;
