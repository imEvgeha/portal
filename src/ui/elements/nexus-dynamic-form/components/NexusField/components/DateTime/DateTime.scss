.nexus-c-date-time-window-picker {
    padding: 0;
    &__start-date,
    &__end-date {
        display: flex;
        align-items: center;
        margin-bottom: 20px;
        & > div:not(.nexus-c-date-time-window-picker__label) {
            flex-grow: 1;
        }
    }
    &__label {
        margin-right: 20px;
    }
}
