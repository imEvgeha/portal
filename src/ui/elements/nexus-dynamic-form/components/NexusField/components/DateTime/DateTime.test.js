import React from 'react';
import {shallow} from 'enzyme';
import DateTime from './DateTime';

describe('DateTime', () => {
    let wrapper = null;
    const props = {
        onChange: () => null,
    };
    beforeAll(() => {
        wrapper = shallow(<DateTime {...props} />);
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render NexusDateTimeWindowPicker when type is dateRange', () => {
        wrapper.setProps({type: 'dateRange'});
        expect(wrapper.find('NexusDateTimeWindowPicker').length).toEqual(1);
    });

    it('should render NexusDateTimePicker when dateType is businessDateTime', () => {
        wrapper.setProps({type: 'datetime', dateType: 'businessDateTime'});
        expect(wrapper.find('NexusDateTimePicker').length).toEqual(1);
    });

    it('should render NexusDatePicker when dateType is regional', () => {
        wrapper.setProps({type: 'datetime', dateType: 'regional'});
        expect(wrapper.find('NexusDatePicker').length).toEqual(1);
    });
});
