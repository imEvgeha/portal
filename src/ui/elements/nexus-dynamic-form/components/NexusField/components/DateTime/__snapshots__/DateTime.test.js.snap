// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTime should match snapshot 1`] = `
<NexusDatePicker
  error=""
  id="-date"
  isClearable={false}
  isLabelHidden={false}
  isReadOnly={false}
  isRequired={false}
  isReturningTime={false}
  isTimestamp={false}
  isWithInlineEdit={false}
  label=""
  onChange={[Function]}
  onConfirm={[Function]}
  value=""
/>
`;
